tuf/ngclient/_internal/trusted_metadata_set.py

```diff
@@ -205,7 +205,7 @@ def update_root(self, data: bytes) -> None:
                 f"Expected 'root', got '{new_root.signed.type}'"
             )
 
-        if self.root is not None:
+        if self._trusted_set.get("root") is not None:
```

Comment:
If the root property is guaranteed to be set after init, do we need this check?

Reply:
We need it because we call `update_root` during init too ... to validate the initial root.
Do you think we need to clarify it somehow?

Reply:
Since you expressed some worries about inconsistently using the 'root' property, the other solution I came up with is this  https://github.com/theupdateframework/tuf/pull/1489/commits/8c172af244c968b221bcbd5398a7a07a6c897408